tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -265,6 +299,25 @@ final class ScalaJSClassEmitter private (
       js.MethodDef(static = false, js.Ident("constructor"), params, body)
     } else {
       val fieldDefs = genFieldDefs(tree)
+      val isFinal = effectivelyFinal(tree)
+      val ctorDefs =
+          tree.memberMethods.map(_.tree).filter(x => isConstructorName(x.name.name))
+      val (otherStatements, args) = ctorDefs match {
+          case defn::Nil if (effectivelyFinal(tree)) =>
+            /* this will introduce an EmptyTree in our body so we have to take
+             * the tail of the body only!
+             */
+            val js.Function(args, bodyBlk) = desugarToConstructor(this,
+                tree.name.name, defn.args, defn.body, js.EmptyTree)
```

Comment:
Pass in `js.Skip()` instead of `js.EmptyTree`, and remove that hack with `bdy.tail`.